File: practica2_839756_842236/c++/instruccionRead.cc
// Autores: Andrei Vlasceanu [839756] & Andres Yubero [842236]

#include "instruccionRead.h"

void InstruccionRead::ejecutar(stack<int> *s)
{
    int x;
    cin >> x;
    s->push(x);
}

void InstruccionRead::listar()
{
    cout << "read" << endl;
}

File: practica2_839756_842236/java/InstruccionRead.java
import java.util.Scanner;
import java.util.Stack;

public class InstruccionRead extends Instruccion {
    @Override
    public void ejecutar(Stack<Integer> s) {
        System.out.print("? ");
        Scanner in = new Scanner(System.in);
        int x = in.nextInt();
        s.push(x);
    }

    @Override
    public void listar() {
        System.out.println("read");
    }
}
